mjs_firmware.ino

```diff
@@ -288,8 +292,14 @@ void getPosition()
   while (millis() - startTime < GPS_TIMEOUT && valid < 10) {
     if (gps.available(gpsSerial)) {
       gps_data = gps.read();
-      if (gps_data.valid.location && gps_data.valid.status && gps_data.status >= gps_fix::STATUS_STD)
+      if (gps_data.valid.location && gps_data.valid.status && gps_data.status >= gps_fix::STATUS_STD) {
         valid++;
+        lat24 = int32_t((int64_t)gps_data.latitudeL() * 32768 / 10000000);
+        lng24 = int32_t((int64_t)gps_data.longitudeL() * 32768 / 10000000);
+      } else {
+        lat24 = 0;
+        lng24 = 0;
+      }
```

Comment:
Why move the packing of data to here? AFAICS the check here `gps_data.valid.location && gps_data.valid.status && gps_data.status >= gps_fix::STATUS_STD)` is the same as the one used when packing, so this change should not affect the result. It might be useful to not duplicate this check, though (and though it would be nicer to store the full latitude/longitude result and then pack them into 2x24 bits in `queueData`, that would also take more memory, so perhaps your current proposal is actually best.

Reply:
@matthijskooijman Since these variables are now globally defined, I set them to zero to explicitly ensure that old GPS data is purged, should the GPS acquisition process fail. If these lines are removed, I believe the previous GPS fix data would be retained. (I could be wrong, obviously :)